classes/abstracts/ActionScheduler_Abstract_Schema.php

```diff
@@ -159,14 +159,25 @@ protected function get_full_table_name( $table ) {
 	public function tables_exist() {
 		global $wpdb;
 
-		$existing_tables = $wpdb->get_col( 'SHOW TABLES' );
 		$expected_tables = array_map(
 			function ( $table_name ) use ( $wpdb ) {
 				return $wpdb->prefix . $table_name;
 			},
 			$this->tables
 		);
 
-		return count( array_intersect( $existing_tables, $expected_tables ) ) === count( $expected_tables );
+		$tables_exist = true;
+
+		foreach ( $expected_tables as $table_name ) {
+			$pattern        = str_replace( '_', '\\_', $table_name );
+			$existing_table = $wpdb->get_var( $wpdb->prepare( 'SHOW TABLES LIKE %s', $pattern ) );
+
+			if ( $existing_table !== $table_name ) {
+				$tables_exist = false;
+				break;
+			}
+		}
+
+		return $tables_exist;
```

Comment:
Instead of looping twice (`array_map` followed by a `foreach`), seems like we could use a single loop? We could also switch to using `esc_like`.

```php
public function tables_exist() {
	global $wpdb;

	foreach ( $this->tables as $table_name ) {
		$table_name     = $wpdb->prefix . $table_name;
		$existing_table = $wpdb->get_var( $wpdb->prepare( 'SHOW TABLES LIKE %s', $wpdb->esc_like( $table_name ) ) );

		if ( $existing_table !== $table_name ) {
			return false;
		}
	}

	return true;
}
```

Reply:
Thanks for the suggestions; I think we're good to go now 👍 
> We could also switch to using esc_like.

Unfortunately, [this breaks HyperDB](https://core.trac.wordpress.org/ticket/57149) 😬
Which is why the `str_replace()` is being performed rather than use `esc_like()`.